### PR TITLE
fix: eslint fixes. Move to PureComponent.

### LINKED ISSUE #12: Refactor the usage of NerdletStateContext, PlatformStateContext, AutoSizer
Current usage falls outside of best practice, and requires refactoring.

### PR DESCRIPTION
**Pull request by tangollama**
- Closes #12

### CODE REVIEW

File: nerdlets/groundskeeper-nerdlet/index.js [hunk 1]
```diff
@@ -675,21 +671,16 @@ export default class Groundskeeper extends React.Component {
...
-      ? `${presentationData.multiversionTable.data.length} apps are running
-              multiple agent versions`
-      : tableState === 'upToDate'
-      ? `${presentationData.currentTable.data.length} apps are up to date
-              with ${upToDateLabel}`
-      : tableState === 'noVersionReported'
-      ? `${presentationData.noVersionsTable.data.length} apps are not
-              reporting agent version data (they may be inactive)`
-      : undefined;
+    let tableBannerText = slaReportKey ? `SLA Report by ${slaReportKey}` : '';
```
Comment (fightingmonk) [inline nerdlets/groundskeeper-nerdlet/index.js]: It's a bit weird to mix a ternary w/ an if/else block, and is tableState guaranteed to not match one of the if clauses below when slaReportKey is non-null? 
Seems better to assign `tableBannerText = ''` and  have the if/else block start w/ `if (slaReportKey) {...} else if (tableState === 'outOfDate') {...`

File: .eslintrc.js [hunk 2]
```diff
@@ -4,8 +4,9 @@ module.exports = {
         "es6": true
     },
     "extends": [
-        "prettier",
-        "prettier/react",
+        'plugin:@newrelic/eslint-plugin-newrelic/react',
+        'plugin:@newrelic/eslint-plugin-newrelic/jest',
+        'plugin:@newrelic/eslint-plugin-newrelic/prettier'
```
Comment (fightingmonk) [inline .eslintrc.js]: Thanks for doing all the style cleanup!

File: nerdlets/groundskeeper-nerdlet/AgentStateLoader.js [hunk 3]
```diff
@@ -22,15 +22,15 @@ const AGENT_NG_QUERY_WEEK = `query($id: Int!) {
     }
 }`;
 
-export default class AgentStateLoader extends React.Component {
+export default class AgentStateLoader extends React.PureComponent {
```
Comment (fightingmonk) [inline nerdlets/groundskeeper-nerdlet/AgentStateLoader.js]: Have you verified that a shallow compare on props and state are always correct about whether to re-render for all of the components? AgentStateLoader should be fine. I'm not entirely confident that the top-level logic always creates new data structures to pass into `props` and `setState` vs. updating existing structures. That could lead `SLAReport` and `Groundskeeper` components to not re-render when they should.

File: nerdlets/groundskeeper-nerdlet/index.js (hunk 1 above)
Comment (tangollama) [reply nerdlets/groundskeeper-nerdlet/index.js]: Great call @fightingmonk. Making that change now.

**Pull request merged**